core/domain/stats_services.py

```diff
@@ -40,50 +40,85 @@ def get_top_unresolved_answers_for_default_rule(exploration_id, state_name):
+    """Gets unresolved answers per exploration for default rule across all
+    states for explorations with ids in exp_ids.
 
-    Note that this method currently returns the counts only for the DEFAULT
-    rule. This should ideally handle all types of unresolved answers.
+    Note that this method currently returns the data only for the DEFAULT rule.
+    This should ideally handle all types of unresolved answers.
 
-    Returns:
-        A dict, keyed by the string '{exp_id}', whose values are the number of
-        unresolved answers that exploration has. Any exp_ids for explorations
-        that don't exist or that have been deleted will be ignored, and not
-        included in the return value.
+    Returns a dict of the following format:
+        {
+          'exp_id_1': {
+            'count': 10 (number of unresolved answers for this exploration),
+            'unresolved_answers': (list of unresolved answers sorted by count)
+              [
+                {'count': 4, 'value': 'answer_1', 'state': 'Introduction'},
+                {'count': 2, 'value': 'answer_2', 'state': 'Introduction'},
+                {'count': 1, 'value': 'answer_3', 'state': 'End'}
+              ]
+          },
+          'exp_id_2': {
+            'count': 20,
+            'unresolved_answers':
+              [
+                {'count': 8, 'value': 'answer_5', 'state': 'Introduction'},
+                {'count': 3, 'value': 'answer_4', 'state': 'Quest'},
+                {'count': 1, 'value': 'answer_6', 'state': 'End'}
+              ]
+          }
+        }
     """
-    explorations = exp_services.get_multiple_explorations_by_id(
-        exp_ids, strict=False)
-
-    # The variable `exploration_states_tuples` is a list of all
-    # (exp_id, state_name) tuples for the given exp_ids.
-    # E.g. - [
-    #   ('eid1', 'Introduction'),
-    #   ('eid1', 'End'),
-    #   ('eid2', 'Introduction'),
-    #   ('eid3', 'Introduction')
-    # ]
-    # when exp_ids = ['eid1', 'eid2', 'eid3'].
-    explorations_states_tuples = [
-        (exp_domain_object.id, state_key)
-        for exp_domain_object in explorations.values()
-        for state_key in exp_domain_object.states
-    ]
+    def _get_explorations_state_tuples_by_ids(exp_ids):
+        """Returns a list of all (exp_id, state_name) tuples for the given
+        exp_ids.
+        E.g. - [
+          ('eid1', 'Introduction'),
+          ('eid1', 'End'),
+          ('eid2', 'Introduction'),
+          ('eid3', 'Introduction')
+        ]
+        when exp_ids = ['eid1', 'eid2', 'eid3']."""
+        explorations = (
+            exp_services.get_multiple_explorations_by_id(exp_ids, strict=False))
+        return [
+            (exp_domain_object.id, state_key)
+            for exp_domain_object in explorations.values()
+            for state_key in exp_domain_object.states
+        ]
+
+    explorations_states_tuples = _get_explorations_state_tuples_by_ids(exp_ids)
```

Comment:
To be consistent, the method should be _get_explorations_state**s**_tuples_by_ids